js/util/urlParser.js

```diff
@@ -25,9 +28,8 @@ var urlParser = {
     if (url.indexOf(' ') === -1 && url.indexOf('.') > 0) {
       return true
     }
-    // a host from the hosts file is also a URL
-    var hostPart = url.replace(/(:|\/).+/, '')
-    return hosts.indexOf(hostPart) > -1
```

Comment:
I don't see this check in `validateDomain`; I think it needs to be added back there so that custom hosts added to the hosts file work. localhost is usually part of the host file, so you should be able to get rid of the localhost check once you add this back.

Reply:
Isn't `isURLMissingProtocol` supposed to check if the URL is just missing the protocol (ie. https://)?
I added a check using the `hosts.js` list to the validateDomain function to ensure that domains specified in the hosts files are treated as valid. Gonna push it as soon as I get the other reviews done.